dashboard/app/models/user.rb

```diff
@@ -619,9 +619,14 @@ def user_level_locked?(script_level, level)
   # script that hasn't yet been passed, starting its search at the last level we submitted
   def next_unpassed_progression_level(script)
     user_levels_by_level = user_levels_by_level(script)
+    user_levels = user_levels_by_level.values.flatten.select do |user_level|
```

Comment:
That userlevel's level used to be in script 142, but was recently removed: https://github.com/code-dot-org/code-dot-org/commit/135b583968fcefe4ed24867c2937bba9d4fc3c96

One benefit to this change is that it will now work properly if your last progress was on a level that has since been removed from the script.